tests/fertiscan/db/test_sub_label.py

```diff
@@ -218,3 +218,38 @@ def test_get_all_sub_label(self):
 
         self.assertEqual(sub_label_data[0][1], self.text_fr)
         self.assertEqual(sub_label_data[1][1], other_fr)
+
+    def test_get_sub_label_json_with_null_values(self):
+        # Insert a sub-label with NULL for text_fr
+        sub_label.new_sub_label(
+            self.cursor,
+            None,  # Null French text
+            self.text_en,  # English text
+            self.label_id,
+            self.sub_type_id,
+            False,
+        )
+        # Insert a sub-label with NULL for text_en
+        sub_label.new_sub_label(
+            self.cursor,
+            self.text_fr,  # French text
+            None,  # Null English text
+            self.label_id,
+            self.sub_type_2_id,
+            False,
+        )
+
+        # Fetch sub-label data as JSON
+        sub_label_data = sub_label.get_sub_label_json(self.cursor, self.label_id)
+
+        # Assert that the 'en' value for the first sub-label is not empty
+        self.assertTrue(sub_label_data[self.type_en]['en'], "Expected 'en' field to have content")
+        
+        # Assert that the 'fr' value for the first sub-label is empty (since it was NULL)
+        self.assertFalse(sub_label_data[self.type_en]['fr'], "Expected 'fr' field to be empty")
```

Comment:
We should verify the length of the array to be 0

Reply:
It's all the same, but done.